File: hw1/java/Main.java
import java.util.*;
import java.lang.*;
import java.io.*;

public class Main{
    public static void main(String[] args){
        int[] a = { 4, 5, 1, 3, 6, 7, 9, 10, 2, 8 };
        int n = 10;
        int k = 4;
        int ans1 = Hw1.randomizedSelect(a, n, k);
        if(Hw1.Checker(a, n, k, ans1)==true){
            System.out.println("correct");
        }
        else{
            System.out.println("incorrect");
        }
        int ans2 = Hw1.deterministicSelect(a, n, k);
        if(Hw1.Checker(a, n, k, ans2)==true){
            System.out.println("correct");
        }
        else{
            System.out.println("incorrect");
        }

    }
}

File: hw1/python/main.py
from hw1 import *

if __name__=="__main__":
    a = [ 4, 5, 1, 3, 6, 7, 9, 10, 2, 8 ]
    n = 10
    k = 4
    ans1 = randomized_select(a, n, k)
    if checker(a, n, k, ans1)==True:
        print('correct')
    else:
        print('incorrect')
    ans2 = deterministic_select(a, n, k)
    if checker(a, n, k, ans2)==True:
        print('correct')
    else:
        print('incorrect')
